System/SpatialPartitionGrid.cpp

```diff
@@ -90,52 +86,44 @@ namespace RTE {
 				for (int y = topLeftCellY; y <= bottomRightCellY; y++) {
 					int cellId = GetCellIdForCellCoords(x, y);
 					m_Cells[team + 1][cellId].push_back(mo.GetID());
-					m_UsedCells.insert(cellId);
+					m_UsedCellIds.insert(cellId);
 				}
 			}
 		}
 	}
 	/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
 	const std::vector<MovableObject *> & SpatialPartitionGrid::GetMOsInBox(const Box &box, int ignoreTeam) const {
-		static std::vector<MovableObject *> s_MOList; // For script interop... :(
+		static std::vector<MovableObject *> s_MOList; // Note - This static vector exists to allow this data to be passed safely to Lua.
 		s_MOList.clear();
 
-		RTEAssert(ignoreTeam >= Activity::NoTeam && ignoreTeam < Activity::MaxTeamCount, "Invalid ignoreTeam given to GetMOsInRect()!");
+		RTEAssert(ignoreTeam >= Activity::NoTeam && ignoreTeam < Activity::MaxTeamCount, "Invalid ignoreTeam given to SpatialPartitioningGrid::GetMOsInBox()!");
 
 		std::unordered_set<MOID> potentialMOIDs;
 
 		Vector topLeft = box.GetCorner();
 		Vector bottomRight = topLeft + Vector(box.GetWidth(), box.GetHeight());
 
-		int topLeftCellX = topLeft.m_X / m_CellSize;
-		int topLeftCellY = topLeft.m_Y / m_CellSize;
-		int bottomRightCellX = bottomRight.m_X / m_CellSize;
-		int bottomRightCellY = bottomRight.m_Y / m_CellSize;
+		int topLeftCellX = topLeft.GetFloorIntX() / m_CellSize;
```

Comment:
This is broken. Integer divide rounds towards 0, whereas we want to just throw away everything after the decimal. As such the cast to int needs to occur after the division, not before.